src/service/asset-api-adapter.js

```diff
@@ -0,0 +1,248 @@
+    *
+    */
+    var upload = function (method, filename, params, options) {
+        validateFilename(filename);
+        // make sure the parameter is clean
+        method = method.toLowerCase();
+        var urlOptions = $.extend({}, serviceOptions, options);
+        if (urlOptions.transport.contentType === 'application/json') {
+            // whitelist the fields that we actually can send to the api
+            params = _pick(params, assetApiParams);
+        } else { // else we're sending form data which goes directly in request body
+            // For multipart/form-data uploads the filename is not set in the URL,
+            // it's getting picked by the FormData field filename.
+            filename = method === 'post' || method === 'put' ? '' : filename;
+        }
+        var url = buildUrl(filename, urlOptions);
+        var createOptions = $.extend(true, {}, urlOptions, { url: url });
+
+        return http[method](params, createOptions);
+    };
+
+    var publicAPI = {
+        /**
+        * Creates a file in the Asset API. The server will return an error if the file already exist,
+        * check first with a list() or a get().
+        *
+        * @param {string} `filename` (Required) Name of the file to create
+        * @param {object} `params` (Optional) Body parameters to send to the Asset API
+        * @param {object} `options` (Optional) Options object to override global options.
+        *
+        */
+        create: function (filename, params, options) {
+            return upload('post', filename, params, options);
+        },
+
+        /**
+        * Gets a filename in the Asset API. This will actually fetch the asset content, to get a list
+        * of the assets in a scope, use list()
+        *
+        * @param {string} `filename` (Required) Name of the file to retrieve
+        * @param {object} `options` (Optional) Options object to override global options.
+        *
+        */
+        get: function (filename, options) {
+            var urlOptions = $.extend({}, serviceOptions, options);
+            var url = buildUrl(filename, urlOptions);
+            var getOptions = $.extend(true, {}, urlOptions, { url: url });
+
+            return http.get(getOptions);
+        },
+
+        /**
+        * Gets the list of the assets in a scope.
+        *
+        * @param {object} `options` (Optional) Options object to override global options. fullUrl determines if the
+        * callback the list of asset URL's instead of the filename only.
+        *
+        */
+        list: function (options) {
+            var dtd = $.Deferred();
+            var me = this;
+            var urlOptions = $.extend({}, serviceOptions, options);
+            var url = buildUrl('', urlOptions);
+            var getOptions = $.extend(true, {}, urlOptions, { url: url });
+            var fullUrl = options.fullUrl;
+
+            if (!fullUrl) {
+                return http.get({}, getOptions);
+            }
+
```

Comment:
i think the order of these checks may not be right. when i call 
```
    aa = new F.service.Asset({
        account: 'acme-simulations-0',
        project: 'epi-1094-team-auth',
        group: group,
        userId: userid,
        transport: {
            contentType: 'application/json'
        }
    });
aa.list();
```
i get a console error "Uncaught TypeError: Cannot read property 'fullUrl' of undefined".